test/e2e/specs/editor/various/writing-flow.spec.js

```diff
@@ -0,0 +1,1004 @@
+		page,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '1' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '2' );
+		await page.keyboard.press( 'ArrowUp' );
+
+		const paragraphBlock = page
+			.locator( 'role=document[name="Paragraph block"i]' )
+			.first();
+		const paragraphRect = await paragraphBlock.boundingBox();
+		const x = paragraphRect.x + ( 2 * paragraphRect.width ) / 3;
+		const y = paragraphRect.y + paragraphRect.height + 1;
+
+		await page.mouse.move( x, y );
+
+		const inserter = page.locator( 'role=button[name="Add block"i]' );
+		await expect( inserter ).toBeVisible();
+
+		const inserterRect = await inserter.boundingBox();
+		const lowerInserterY = inserterRect.y + ( 2 * inserterRect.height ) / 3;
+
+		await page.mouse.click( x, lowerInserterY );
+		await page.keyboard.type( '3' );
+
+		await expect.poll( editor.getEditedPostContent )
+			.toBe( `<!-- wp:paragraph -->
+<p>1</p>
+<!-- /wp:paragraph -->
+
+<!-- wp:paragraph -->
+<p>23</p>
+<!-- /wp:paragraph -->` );
+	} );
+
+	test( 'should not have a dead zone above an aligned block', async ( {
+		editor,
+		page,
+	} ) => {
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '1' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '/image' );
+		await page.keyboard.press( 'Enter' );
+		await editor.clickBlockToolbarButton( 'Align' );
+
+		const wideButton = page.locator(
+			'role=menuitemradio[name="Wide width"i]'
+		);
+		await wideButton.click();
+
+		// Focus the block content
+		await page.keyboard.press( 'Tab' );
+
+		// Select the previous block.
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.press( 'ArrowUp' );
+
+		// Confirm correct setup.
+		await expect.poll( editor.getEditedPostContent )
+			.toBe( `<!-- wp:paragraph -->
+<p>1</p>
+<!-- /wp:paragraph -->
+
+<!-- wp:image {"align":"wide"} -->
+<figure class="wp-block-image alignwide"><img alt=""/></figure>
+<!-- /wp:image -->` );
+
+		const paragraphBlock = page.locator(
+			'role=document[name="Paragraph block"i]'
+		);
+
+		// Find a point outside the paragraph between the blocks where it's
+		// expected that the sibling inserter would be placed.
+		const paragraphRect = await paragraphBlock.boundingBox();
+		const x = paragraphRect.x + ( 2 * paragraphRect.width ) / 3;
+		const y = paragraphRect.y + paragraphRect.height + 1;
+
+		await page.mouse.move( x, y );
+
+		const inserter = page.locator( 'role=button[name="Add block"i]' );
+		await expect( inserter ).toBeVisible();
+
+		// Find the space between the inserter and the image block.
+		const inserterRect = await inserter.boundingBox();
+		const lowerInserterY = inserterRect.y + ( 2 * inserterRect.height ) / 3;
+
+		// Clicking that in-between space should select the image block.
+		await page.mouse.click( x, lowerInserterY );
+
+		await expect(
+			page.locator( 'role=document[name="Block: Image"i]' )
+		).toHaveClass( /is-selected/ );
```

Comment:
Well, I'm not sure if class names should be considered public API and worth testing for? Would `toBeFocused` work here?

Reply:
We consider blocks to be selected even if we focus on inner elements. This was the reason I had to use the class.